src/common/helper.js

```diff
@@ -454,31 +461,49 @@ async function getMembers (token, handles) {
 
 /**
  * Function to get project by id
- * @param {String} token the user request token
+ * @param {Object} currentUser the user who perform this operation
  * @param {Number} id project id
  * @returns the request result
  */
-async function getProjectById (token, id) {
+async function getProjectById (currentUser, id) {
+  let token
+  if (currentUser.isBookingManager || currentUser.isMachine) {
+    const m2mToken = await getM2Mtoken()
+    token = `Bearer ${m2mToken}`
+  } else {
+    token = currentUser.jwtToken
+  }
   const url = `${config.TC_API}/projects/${id}`
-  const res = await request
-    .get(url)
-    .set('Authorization', token)
-    .set('Content-Type', 'application/json')
-    .set('Accept', 'application/json')
-  localLogger.debug({ context: 'getProjectById', message: `response body: ${JSON.stringify(res.body)}` })
-  return _.pick(res.body, ['id', 'name'])
+  try {
+    const res = await request
+      .get(url)
+      .set('Authorization', token)
+      .set('Content-Type', 'application/json')
+      .set('Accept', 'application/json')
+    localLogger.debug({ context: 'getProjectById', message: `response body: ${JSON.stringify(res.body)}` })
+    return _.pick(res.body, ['id', 'name'])
+  } catch (err) {
+    console.log(err)
```

Comment:
@imcaizheng let's use `localLogger` or remove `console.log`.

Reply:
@maxceem  Yeah, please help me remove the redundant code. I forgot to delete it;)

Reply:
@imcaizheng you may just push one more commit to `imcaizheng:team-service-m2m-internal-api-calls`, and create a new PR. I would merge it.

Or you can remove it in any other PR you are currently working on. Just pull the latest changes from `dev` branch, as I've already merged this PR.

Reply:
@maxceem PR created https://github.com/topcoder-platform/taas-apis/pull/75

Reply:
thank you @imcaizheng. Merged.